<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>mutli_iodev_params</class>
 <widget class="QWidget" name="mutli_iodev_params">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>416</width>
    <height>360</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Maximum" vsizetype="Maximum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QFrame" name="type_frame">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="frameShape">
        <enum>QFrame::StyledPanel</enum>
       </property>
       <property name="frameShadow">
        <enum>QFrame::Raised</enum>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout_2">
        <item>
         <widget class="QPushButton" name="tbTcp">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="multi_iodev_params.qrc">
            <normaloff>:/multi_iodev_params/tcp</normaloff>:/multi_iodev_params/tcp</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">type_connection</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="tbUdp">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="multi_iodev_params.qrc">
            <normaloff>:/multi_iodev_params/udp</normaloff>:/multi_iodev_params/udp</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">type_connection</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="tbSerial">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="multi_iodev_params.qrc">
            <normaloff>:/multi_iodev_params/serial</normaloff>:/multi_iodev_params/serial</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">type_connection</string>
          </attribute>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="tbFtdi">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="multi_iodev_params.qrc">
            <normaloff>:/multi_iodev_params/usb</normaloff>:/multi_iodev_params/usb</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="autoExclusive">
           <bool>true</bool>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">type_connection</string>
          </attribute>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="tbApply">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="multi_iodev_params.qrc">
            <normaloff>:/multi_iodev_params/apply.png</normaloff>:/multi_iodev_params/apply.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="tbUndo">
          <property name="text">
           <string/>
          </property>
          <property name="icon">
           <iconset resource="multi_iodev_params.qrc">
            <normaloff>:/multi_iodev_params/undo.png</normaloff>:/multi_iodev_params/undo.png</iconset>
          </property>
          <property name="iconSize">
           <size>
            <width>32</width>
            <height>32</height>
           </size>
          </property>
          <property name="flat">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="gb_net_params">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Параметры подключения</string>
       </property>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="host_addr_label">
          <property name="text">
           <string>Адрес</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1" colspan="2">
         <widget class="QLineEdit" name="host_addr">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="host_port_label">
          <property name="text">
           <string>Порт</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="host_port">
          <property name="maximum">
           <number>65535</number>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>0</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="gb_serial_device">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="title">
        <string>Параметры устройства</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <property name="spacing">
           <number>6</number>
          </property>
          <property name="sizeConstraint">
           <enum>QLayout::SetDefaultConstraint</enum>
          </property>
          <property name="leftMargin">
           <number>4</number>
          </property>
          <property name="topMargin">
           <number>4</number>
          </property>
          <property name="rightMargin">
           <number>4</number>
          </property>
          <property name="bottomMargin">
           <number>4</number>
          </property>
          <item>
           <widget class="QComboBox" name="dev_name">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Maximum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="editable">
             <bool>true</bool>
            </property>
            <property name="maxVisibleItems">
             <number>8</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="label_5">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Скорость</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cbBaud">
            <property name="sizePolicy">
             <sizepolicy hsizetype="MinimumExpanding" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="editable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="cbDataSize">
            <item>
             <property name="text">
              <string>8 bit</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>7 bit</string>
             </property>
            </item>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="sbStopBits">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="frame">
             <bool>true</bool>
            </property>
            <property name="suffix">
             <string> стоп бит</string>
            </property>
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>2</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QGroupBox" name="gb_serial_parity">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>Четность</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_2">
             <item row="0" column="0">
              <widget class="QRadioButton" name="rb_parity_none">
               <property name="text">
                <string>Нет</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QRadioButton" name="rb_parity_even">
               <property name="text">
                <string>Чёт</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QRadioButton" name="rb_parity_odd">
               <property name="text">
                <string>Нечет</string>
               </property>
               <property name="checked">
                <bool>false</bool>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout">
            <item>
             <widget class="QGroupBox" name="gb_flow_serial">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title">
               <string>Контроль</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_8">
               <property name="spacing">
                <number>6</number>
               </property>
               <property name="leftMargin">
                <number>9</number>
               </property>
               <property name="topMargin">
                <number>9</number>
               </property>
               <property name="rightMargin">
                <number>9</number>
               </property>
               <property name="bottomMargin">
                <number>9</number>
               </property>
               <item>
                <widget class="QRadioButton" name="rb_flow_serial_none">
                 <property name="text">
                  <string>нет</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb_flow_serial_hardware">
                 <property name="text">
                  <string>Аппаратный</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb_flow_serial_software">
                 <property name="text">
                  <string>Программный</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="gb_flow_ftdi">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="title">
               <string>Контроль</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <property name="spacing">
                <number>6</number>
               </property>
               <property name="leftMargin">
                <number>9</number>
               </property>
               <property name="topMargin">
                <number>9</number>
               </property>
               <property name="rightMargin">
                <number>9</number>
               </property>
               <property name="bottomMargin">
                <number>9</number>
               </property>
               <item>
                <widget class="QRadioButton" name="rb_flow_none">
                 <property name="text">
                  <string>нет</string>
                 </property>
                 <property name="checked">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb_flow_rts_cts">
                 <property name="text">
                  <string>RTS+CTS</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb_flow_dtr_dsr">
                 <property name="text">
                  <string>DTR+DSR</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QRadioButton" name="rb_flow_xon_xof">
                 <property name="text">
                  <string>XON XOF</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>0</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer_4">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>0</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <spacer name="horizontalSpacer_3">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>host_addr</tabstop>
  <tabstop>host_port</tabstop>
  <tabstop>cbBaud</tabstop>
  <tabstop>cbDataSize</tabstop>
  <tabstop>sbStopBits</tabstop>
  <tabstop>rb_parity_none</tabstop>
  <tabstop>rb_parity_even</tabstop>
  <tabstop>rb_parity_odd</tabstop>
  <tabstop>rb_flow_serial_none</tabstop>
  <tabstop>rb_flow_serial_hardware</tabstop>
  <tabstop>rb_flow_serial_software</tabstop>
  <tabstop>rb_flow_none</tabstop>
  <tabstop>rb_flow_rts_cts</tabstop>
  <tabstop>rb_flow_dtr_dsr</tabstop>
  <tabstop>rb_flow_xon_xof</tabstop>
 </tabstops>
 <resources>
  <include location="multi_iodev_params.qrc"/>
 </resources>
 <connections/>
 <buttongroups>
  <buttongroup name="type_connection"/>
 </buttongroups>
</ui>
